posthog/models/filters/mixins/simplify.py

```diff
@@ -38,8 +40,15 @@ def simplify(self: T, team: "Team", **kwargs) -> T:
         properties = self._simplify_properties(team, result.properties, **kwargs)  # type: ignore
         if getattr(result, "aggregation_group_type_index", None) is not None:
             properties.append(self._group_set_property(cast(int, result.aggregation_group_type_index)))  # type: ignore
+            new_group_props.append(self._group_set_property(cast(int, result.aggregation_group_type_index)))  # type: ignore
 
-        return result.with_data({**updated_entities, "properties": properties,})
+        combined_group = {}
+        if new_group_props and result.property_groups_to_dict():
```

Comment:
I meant, why do we choose the pre-simplified version of properties, and not post-simplified? (That means you can also get rid of stuff on Line 43)

Reply:
.. and the cohort simplification looks tricky, since that recursively generates properties.

I'm guessing for now cohort property group simplification is out of scope?

Reply:
oh I'm keeping the logic separate because I didn't want some weird situation where if we use post-simplified properties, it will include all other properties. The logic for simplifying properties isn't using a fresh list it's just directly appending to `filter.properties` so if i used properties instead of `new_gorup_props` i _could_ end up having repeats